collector/src/bin/rustc-perf-collector/execute.rs

```diff
@@ -262,6 +258,265 @@ lazy_static! {
     };
 }
 
+pub trait Processor {
+    fn profiler(&self) -> Profiler;
+
+    fn process_output(&mut self, name: &str, cwd: &Path, build_kind: BuildKind, run_kind: RunKind,
+                      run_kind_str: &str, patch: Option<&Patch>, output: process::Output)
+                      -> Result<bool, Error>;
+
+    fn finish_build_kind(&mut self, _build_kind: BuildKind) {}
+}
+
+pub struct MeasureProcessor {
+    clean_stats: Vec<Vec<Stat>>,
+    nll_stats: Vec<Vec<Stat>>,
+    base_incr_stats: Vec<Vec<Stat>>,
+    clean_incr_stats: Vec<Vec<Stat>>,
+    patched_incr_stats: Vec<(Patch, Vec<Vec<Stat>>)>,
+
+    pub collected: CollectedBenchmark,
+}
+
+impl MeasureProcessor {
+    pub fn new(name: &str) -> Self {
+        // Check we have `perf` available.
+        let has_perf = Command::new("perf").output().is_ok();
+        assert!(has_perf);
+
+        MeasureProcessor {
+            clean_stats: Vec::new(),
+            nll_stats: Vec::new(),
+            base_incr_stats: Vec::new(),
+            clean_incr_stats: Vec::new(),
+            patched_incr_stats: Vec::new(),
+
+            collected: CollectedBenchmark {
+                name: name.to_string(),
+                runs: Vec::new(),
+            },
+        }
+    }
+}
+
+impl Processor for MeasureProcessor {
+    fn profiler(&self) -> Profiler {
+        Profiler::PerfStat
+    }
+
+    fn process_output(&mut self, _name: &str, _cwd: &Path, _build_kind: BuildKind,
+                      run_kind: RunKind, _run_kind_str: &str, patch: Option<&Patch>,
+                      output: process::Output) -> Result<bool, Error> {
```

Comment:
Can we refactor all of these arguments into some struct? Not sure what it would be called, but I think that would aid readability especially since we don't use many of them.